# assignment 2

### by alli kirste

in this assignment, i'm observing the distribution of public space across urban centers. the file that I selected provides my with data for every country, I eventually mostly hone in on the united states. the dataset comes from NASA's Socioeconomic Data and Applications Center. i'm really interested in the distribution of public space as it relates to residents' well-being, environmental co-benefits, and transportation efficiency. 

first, I will import pandas and geopandas as pd and gpd. I think for the sake of this file I will just be using pandas. 

In [1]:
import pandas as pd

In [2]:
import geopandas as gpd

next, I define the dataframe as "public_access_table_2023", which is where the data that I will be looking at is located 

In [3]:
df = pd.read_csv('public_access_table_2023.csv')

below are the first 5 data entries, all of which describe entries in Afghanistan

In [4]:
df.head()

OID_ CTR_MN_ISO    CTR_MN_NM    UC_NM_MN   UC_NM_LST       AREA   AREA_PS  \
0     1        AFG  Afghanistan         NaN           -   4.988087  0.233059   
1     2        AFG  Afghanistan     Ghurian     Ghurian   5.985313  0.444520   
2     3        AFG  Afghanistan         NaN           -   1.995363  0.086970   
3     4        AFG  Afghanistan  Toraghundi  Toraghundi   6.984317  0.318201   
4     5        AFG  Afghanistan       Herat       Herat  59.852506  8.659947   

   SDG1171pct      Total_POP    Access_POP   Pct_Access  Shape_Length  \
0    4.672304    2735.464226  -9999.000000 -9999.000000      0.128493   
1    7.426841     929.483667    117.249717    12.614500      0.105675   
2    4.358591      96.640578  -9999.000000 -9999.000000      0.061115   
3    4.555930   14976.737288   2352.677170    15.708877      0.145352   
4   14.468812  317854.237989  61207.232646    19.256384      0.518800   

   Shape_Area  
0    0.000491  
1    0.000586  
2    0.000197  
3    0.000692  
4    0.005864

Checking the type of the dataframe

In [5]:
type(df)

pandas.core.frame.DataFrame

Renaming the dataframe

In [7]:
public_access = df

In [9]:
print ('public_access')

public_access


checking that the rename worked! 

In [10]:
public_access.head()

OID_ CTR_MN_ISO    CTR_MN_NM    UC_NM_MN   UC_NM_LST       AREA   AREA_PS  \
0     1        AFG  Afghanistan         NaN           -   4.988087  0.233059   
1     2        AFG  Afghanistan     Ghurian     Ghurian   5.985313  0.444520   
2     3        AFG  Afghanistan         NaN           -   1.995363  0.086970   
3     4        AFG  Afghanistan  Toraghundi  Toraghundi   6.984317  0.318201   
4     5        AFG  Afghanistan       Herat       Herat  59.852506  8.659947   

   SDG1171pct      Total_POP    Access_POP   Pct_Access  Shape_Length  \
0    4.672304    2735.464226  -9999.000000 -9999.000000      0.128493   
1    7.426841     929.483667    117.249717    12.614500      0.105675   
2    4.358591      96.640578  -9999.000000 -9999.000000      0.061115   
3    4.555930   14976.737288   2352.677170    15.708877      0.145352   
4   14.468812  317854.237989  61207.232646    19.256384      0.518800   

   Shape_Area  
0    0.000491  
1    0.000586  
2    0.000197  
3    0.000692  
4    0.005864

In [12]:
public_access.value_counts()

OID_   CTR_MN_ISO  CTR_MN_NM    UC_NM_MN           UC_NM_LST          AREA       AREA_PS    SDG1171pct  Total_POP      Access_POP      Pct_Access    Shape_Length  Shape_Area
2      AFG         Afghanistan  Ghurian            Ghurian            5.985313   0.444520   7.426841    929.483667      117.249717      12.614500    0.105675      0.000586      1
8876   ITA         Italy        Novara             Novara             20.001782  2.825734   14.127412   87921.098975    87137.554406    99.108809    0.254438      0.002300      1
8867   ISR         Israel       Netanya            Netanya            60.804195  11.286650  18.562288   149253.887623   129594.125384   86.827973    0.398903      0.005823      1
8868   ISR         Israel       Baqa al Gharbiyya  Baqa al Gharbiyya  16.951181  2.166824   12.782733   30117.048903    10306.721521    34.222216    0.199557      0.001625      1
8869   ISR         Israel       Beer Sheva         Beer Sheva         38.881745  10.953616  28.171616   117315

In [13]:
public_access.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13135 entries, 0 to 13134
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   OID_          13135 non-null  int64  
 1   CTR_MN_ISO    13135 non-null  object 
 2   CTR_MN_NM     13135 non-null  object 
 3   UC_NM_MN      12262 non-null  object 
 4   UC_NM_LST     13135 non-null  object 
 5   AREA          13135 non-null  float64
 6   AREA_PS       13135 non-null  float64
 7   SDG1171pct    13135 non-null  float64
 8   Total_POP     13135 non-null  float64
 9   Access_POP    13135 non-null  float64
 10  Pct_Access    13135 non-null  float64
 11  Shape_Length  13135 non-null  float64
 12  Shape_Area    13135 non-null  float64
dtypes: float64(8), int64(1), object(4)
memory usage: 1.3+ MB


In [71]:
#here is a sample of 5 of the countries in this database
public_access['CTR_MN_ISO'].sample(5)

4535     FRA
7464     IND
2480     CHN
12437    USA
1705     CHN
Name: CTR_MN_ISO, dtype: object

In [35]:
#trying to cut out some of the more erroneous information, focusing on the name of the country, the name of the urban center
#the size of the population with access to the public space, the size of the public space within each urban center (in square kilometers), 
#and the percent of the population with access to the public space
desired_columns = ['CTR_MN_ISO','UC_NM_MN','Access_POP','AREA_PS','Pct_Access']

In [18]:
print (desired_columns)

['CTR_MN_ISO', 'Access_POP', 'AREA_PS', 'Pct_Access']


In [31]:
public_access[desired_columns].sample(5)

CTR_MN_ISO           UC_NM_MN    Access_POP      AREA_PS   Pct_Access
8661         IRN             Alvand  2.456620e+04     2.122745    52.881237
6177         IND            dehgaon -9.999000e+03 -9999.000000 -9999.000000
2486         CHN                浠水县  9.023863e+02     0.796604     0.787438
3962         ESP  Palma de Mallorca  4.213840e+05    20.881327    96.179839
12308        USA           Portland  1.520235e+06   241.630973    87.705121

In [69]:
#creating a new dataframe
public_access_trimmed = public_access[desired_columns].copy()

In [37]:
#checking it
public_access_trimmed

CTR_MN_ISO    UC_NM_MN    Access_POP   AREA_PS   Pct_Access
0            AFG         NaN  -9999.000000  0.233059 -9999.000000
1            AFG     Ghurian    117.249717  0.444520    12.614500
2            AFG         NaN  -9999.000000  0.086970 -9999.000000
3            AFG  Toraghundi   2352.677170  0.318201    15.708877
4            AFG       Herat  61207.232646  8.659947    19.256384
...          ...         ...           ...       ...          ...
13130        ZWE   Marondera   3005.662121  0.483741    11.396391
13131        ZWE      Murewa   1049.279741  0.227534    13.665305
13132        ZWE   Checheche  -9999.000000  0.141944 -9999.000000
13133        ZWE         NaN  -9999.000000  0.029564 -9999.000000
13134        ZWE      Mutare  13137.362411  1.395680    16.313614

[13135 rows x 5 columns]

In [38]:
#looking at the access to public spaces specifically in the US. I think it would be interesting to run a comparison of a few countries to understand
#whether or not public space is distributed equitably internationally. Or even at a US level, figuring out which states have the largest/smallest amounts of public space
#and the most/fewest people that can access it
public_access_trimmed[public_access_trimmed.CTR_MN_ISO == 'USA']

CTR_MN_ISO       UC_NM_MN     Access_POP    AREA_PS  Pct_Access
12248        USA       Honolulu  446071.877762  46.955676   86.399982
12249        USA    Santa Maria   84238.236972  12.213078   68.320696
12250        USA       Monterey   62493.551939  23.595040   92.157546
12251        USA  Santa Barbara   94854.030861  13.519621   82.466867
12252        USA     Santa Cruz   95191.996514  28.606683   95.342361
...          ...            ...            ...        ...         ...
12567        USA         Nashua   49351.366483   8.923710   81.054325
12568        USA         Lowell   96526.386322  10.896601   89.652276
12569        USA     Manchester   73431.325863  13.360204   84.529094
12570        USA       Lawrence   92090.813842  10.217717   80.631907
12571        USA       Portland   82101.593654  19.857901   92.245236

[324 rows x 5 columns]

In [66]:
# here is the same public access data, but for the philippines
public_access_trimmed[public_access_trimmed.CTR_MN_ISO == 'PHL']

CTR_MN_ISO      UC_NM_MN     Access_POP   AREA_PS  Pct_Access
10827        PHL         Laoag   19575.925340  0.951978   28.209797
10828        PHL         Vigan   16107.763276  1.407175   19.624387
10829        PHL  San Fernando   33926.094790  1.623295   27.314123
10830        PHL      Lingayen   21620.145783  1.406679   21.087840
10831        PHL       Dagupan   91417.619566  5.961701   15.224787
...          ...           ...            ...       ...         ...
10915        PHL         Digos   10469.505640  1.157872   15.798768
10916        PHL         Toril  159919.793962  4.730891   44.042672
10917        PHL    Davao City  538234.671682  7.776624   64.165884
10918        PHL        Panabo   13991.280768  0.672602   16.292526
10919        PHL         Tagum  143606.060814  4.571430   55.814748

[93 rows x 5 columns]

In [67]:
#defining the variable
Pct_Access = public_access['Pct_Access'].value_counts()

In [53]:
Pct_Access

Pct_Access
-9999.000000    4262
 100.000000       56
 25.786044         1
 9.124525          1
 58.558558         1
                ... 
 99.977118         1
 99.374454         1
 99.578209         1
 99.562786         1
 16.313614         1
Name: count, Length: 8819, dtype: int64

In [56]:
# Not sure how/if I can delete the percent access that are errors. I also want to sort the data by percent access ascending but only for the US

In [68]:
#trimming to only look at the United States
usa_public_access = public_access_trimmed[public_access_trimmed.CTR_MN_ISO == 'USA'].copy()

In [59]:
usa_public_access

CTR_MN_ISO       UC_NM_MN     Access_POP    AREA_PS  Pct_Access
12248        USA       Honolulu  446071.877762  46.955676   86.399982
12249        USA    Santa Maria   84238.236972  12.213078   68.320696
12250        USA       Monterey   62493.551939  23.595040   92.157546
12251        USA  Santa Barbara   94854.030861  13.519621   82.466867
12252        USA     Santa Cruz   95191.996514  28.606683   95.342361
...          ...            ...            ...        ...         ...
12567        USA         Nashua   49351.366483   8.923710   81.054325
12568        USA         Lowell   96526.386322  10.896601   89.652276
12569        USA     Manchester   73431.325863  13.360204   84.529094
12570        USA       Lawrence   92090.813842  10.217717   80.631907
12571        USA       Portland   82101.593654  19.857901   92.245236

[324 rows x 5 columns]

In [72]:
#here is a sorted chart showing the urban areas with the lowest percent access to public spaces all the way to highest percent.
#I would also be interested in perhaps mapping these "urban centers" with public spaces that they identified
usa_public_access.sort_values(by = 'Pct_Access', ascending = True)

CTR_MN_ISO          UC_NM_MN     Access_POP    AREA_PS  Pct_Access
12300        USA  Lake Havasu City    6756.775299   8.328210   12.166804
12440        USA    Port St. Lucie   20773.654837  15.296814   15.644214
12376        USA        Springdale   11791.493994   7.399025   17.171895
12415        USA       Spring Hill   13429.431354  11.419182   18.268936
12410        USA          Columbus   31477.695811  27.516467   19.962895
...          ...               ...            ...        ...         ...
12548        USA             Dover   53961.624010  15.198764   98.549855
12282        USA             Davis   75569.019636   8.156081   98.679827
12429        USA         Rochester   77845.962161  25.002692   98.768023
12318        USA         Anchorage  126854.407701  22.884413   99.336394
12518        USA          Leesburg   81063.710808  12.224534   99.949272

[324 rows x 5 columns]

so there are 324 urban areas in the united states that are represented in this dataset, with the urban center with the lowest percent access to public space being Lake Havasu City in Arizona and the highest with nearly 100% being Leesburg in Virgina

In [73]:
#some info about los angeles! 
usa_public_access[usa_public_access.UC_NM_MN == 'Los Angeles']

CTR_MN_ISO     UC_NM_MN    Access_POP      AREA_PS  Pct_Access
12260        USA  Los Angeles  8.578123e+06  1388.072463   59.616598

so, about 60% of LA's population live within 400 meters of the public space polygon identified by nasa's sedac team